### PR TITLE
finshed assessment

### PR DESCRIPTION
**Pull request by ResetReesi**
# Submitting Week 5 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [ 😢] JavaScript Coding Challenges
- [ 😢 ] Ruby Coding Challenges
- [ ✅] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:
When it came to ruby I felt like I forgot everything lol. I need more practice with javascript. This weekend I took the time to go through my vocabulary with my husband testing me. I feel a lot better now that with understanding everything. I realize this weekend it like to learn through games and creativity. Thanks for taking the time to go over my my assessment.

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -10,43 +10,207 @@
...
+
+const secretMessage = (array) => {
+ return secretCodeWord1.replace(/[aeiou]/gi, "4" )
+
+}
+return  secretCodeWord1
+
+console.log(secretMessage(secretCodeWord1))
+// // console.log(secretMessage(secretCodeWord2))
+// // console.log(secretMessage(secretCodeWord1))
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:82]: I like what you are trying to do here, I know JS is hard. 

That said this is the second week in a row I am giving you this feedback about functions and their syntax. **_Let's chat about this if you need help._** 

Secret message on 61 and secret message on 74 are in direct conflict with one another. 

There should only be one function "secretMessage" not two. It should be reusable.

File: code-challenges.test.js [hunk 2]
```diff
@@ -10,43 +10,207 @@
...
+    //   This was a hard one 🥹.. 💔 (still borken any reason why?) *also is concole.log needed if you run yarn jest.
+
+const wordleGame = (value) => {
+
+  return array.filter((vale) => 
+    value.tolowercase().includes(variable))
+ }
+
+console.log(wordleGame())
+
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:147]: One function much better! 

Variable is not being declared anywhere. What you are looking for is a tool called a parameter.

File: code-challenges.test.js [hunk 3]
```diff
@@ -10,43 +10,207 @@
...
+    //  Expected output: false
+    const hand4 = [7, 2, 7, 2, 7]
+    //  Expected output: true
+expect(GamePoker(hand1)).toequal(expect.(true))
+expect(GamePoker(hand2)).toequal(expect.(false))
+expect(GamePoker(hand3)).toequal(expect.(false))
+expect(GamePoker(hand4)).toequal(expect.(true))
+
+    })
+})
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:171]: Remember functions are lowerCase camelCase.  
You don't want the extra expect on each line. 
Other than that this is great!

File: interview-questions.md [hunk 4]
```diff
@@ -10,42 +10,81 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+
 5. What are primary keys? Why are they important?
 
 Your answer:
+A primary key is a like a house key. Its a unigue key for every record in the the data base. A primary key is personalized id. When you think of a primary key think of your telephone number. You are the only perosn with the telephone number that you have. no on else can have the same number as you. Your id can also be a unique identifier, No on has that exact id. Its meant for just you. How to identify you. They even help make up a relational data base. They are important because it is a way to help you keep track of data and access the data quick and easy.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:59]: Great analogy!

File: interview-questions.md [hunk 5]
```diff
@@ -10,42 +10,81 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+
+  
 
 Researched answer:
 
+During my research I found that relational database has principles that should be followed meaning all items in a cloumn must have the same type of information for each user. In each column that you create it must have the same amount of rows and tables. Everything has to sync and look as if the format workd for everyone. for example when you go to instagram every post looks the same. It has a user name, a picture than a story of choice. Every instagram user has the same structure but the information for each user will be different. By using this format it will help identify each row because each user will have it own primary key. Finding out that there are alot more databases then I thought was pretty cool. I did not know that there was a non relational database and thet there are 4 types of non relational databases. what this data base is basically a database used for a huge amount of data. For example twiiter,facebook and other big companies that have millions of users use this type of data base. It's mainly used because it usually performs at a fatser level since there are not alot og tables to sort from in order to deliver what your looking for.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:54]: Very good info, but you refer to "a format" a lot without defining what that format is. I wanna know about this format it sounds cool!

File: interview-questions.md [hunk 6]
```diff
@@ -10,42 +10,81 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 3. What is Ruby on Rails?
 
 Your answer:
 
+Ruby on rails is basically everything you need to be an amazing developer. It has everything you need to build a full-stack framework. Its an all in one exspense paid trip if you aske me. It provides everything you need to build a font end and back end application. I remember in class we made a txt based game in ruby and I thoight that was so unique.
+
+
 Researched answer:
 
+Reasearch goes into more of the perks of using ruby and rails. Ruby and rails is really just Ruby written in a ceratin way. They coinside with each other. They are like family members. There are some differences that I think are important to know. For example evrything is ruby is an object. Ruby on rails is mainly uses to biuld web application and use server side scripting which means designing websites that processes a user request. This will send the request to the home server and return the data the usere requested. It's also used to develop interface scripts which leps on the fron end of things.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:42]: Dope!

File: ruby-challenges.rb [hunk 7]
```diff
@@ -1,32 +1,110 @@
...
+    array.select do |value|
+        if value.include?(variable)
+            value 
+        end
+    end
+end
+
+
+    p letter_checker(beverages_array, letter_o)
+        output: ["coffee", "soda water"]
```
Comment (WalkerTexas-coder) [inline ruby-challenges.rb:29]: Very cool!

File: ruby-challenges.rb [hunk 8]
```diff
@@ -1,32 +1,110 @@
...
+    def pedal_faster(number)
+        @current_speed += number
+    end
+    def brake(number)
+        if ((@current_speed - number) <= 0)
+            @current_speed = 0
+        else 
+            @current_speed -= number
+        end
+    end
```
Comment (WalkerTexas-coder) [inline ruby-challenges.rb:88]: Excellent work!

**Review by WalkerTexas-coder — APPROVED** [review]
Brook your interview answers are killer! Getting better each week. The Ruby looks really good too. Most of the issues will get smoothed over with practice. 

Javascript is a good attempt but like you pointed out there are some fundamental concepts missing. This is going to be a blocker moving forward for you if you don't take care of it soon. 

Please go back to the functions syllabus page and the lecture that is linked and go through it and do the challenges. 
Please go back to the jest lecture and go through it and do the challenges. 
I also recommend going through the challenges from later in the javascript unit 2 section! 
Practice is the only thing that will move this forward.

File: code-challenges.test.js (hunk 1 above)
Comment (ResetReesi) [reply code-challenges.test.js:82]: Please help me Austin thanks

File: interview-questions.md (hunk 5 above)
Comment (ResetReesi) [reply interview-questions.md:54]: sounds good

**Pull request merged**